v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/settings_manager.py

```diff
@@ -62,7 +63,16 @@ def __upgrade_obsolete_settings(self, settings: dict):
         }.items():
             value = settings.get(setting_key)
             if value == obsolete:
-                self.__log(f"Changed obsolete setting: {obsolete} to {new}.")
+                self.__log(f"Changed setting ({setting_key}): {obsolete} to {new}.")
+                settings.update({setting_key: new})
+
+        setting_key = "input_text.fm_host_url"
+        for obsolete, new in {
```

Comment:
This is a code duplication of lines 60-67. 
I think we should create a helper function for this.

Reply:
Indeed, i'll figure out how to deal with this.

Reply:
As it is no longer a loop it 's not a duplicate. Further this is code that can/should be removed after all users have upgraded so i don't want to put much time into it any more.